src/nf/nf_conv1d_layer_submodule.f90

```diff
@@ -0,0 +1,183 @@
+submodule(nf_conv1d_layer) nf_conv1d_layer_submodule
+
+  use nf_activation, only: activation_function
+  use nf_random, only: random_normal
+
+  implicit none
+
+contains
+
+  module function conv1d_layer_cons(filters, kernel_size, activation) result(res)
+    implicit none
```

Comment:
```suggestion
```
These `implicit none` are not needed, as already covered by the `implicit none` of the submodule.